includes/modules/pages/login/header_php.php

```diff
@@ -145,25 +145,8 @@
 // Check for PayPal express checkout button suitability:
 $paypalec_enabled = (defined('MODULE_PAYMENT_PAYPALWPP_STATUS') && MODULE_PAYMENT_PAYPALWPP_STATUS == 'True' && defined('MODULE_PAYMENT_PAYPALWPP_ECS_BUTTON') && MODULE_PAYMENT_PAYPALWPP_ECS_BUTTON == 'On');
 // Check for express checkout button suitability (must have cart contents, value > 0, and value < 10000USD):
-if ($currencies->is_set('USD')) {
-    $amount = $currencies->value($_SESSION['cart']->total, true, 'USD');
-} else {
-    require_once DIR_FS_CATALOG . DIR_WS_FUNCTIONS . 'functions_exchange_rates.php';
-    $quote_function = 'quote_' . CURRENCY_SERVER_PRIMARY . '_currency';
-    if (function_exists($quote_function)) {
-        $rate = $quote_function('USD');
-    }
-    if (empty($rate) && !empty(CURRENCY_SERVER_BACKUP)) {
-        $quote_function = 'quote_' . CURRENCY_SERVER_BACKUP . '_currency';
-        if (function_exists($quote_function)) {
-            $rate = $quote_function('USD');
-        }
-    }
-    $multiplier = (defined('CURRENCY_UPLIFT_RATIO') && (int)CURRENCY_UPLIFT_RATIO != 0) ? CURRENCY_UPLIFT_RATIO : 0;
-    $amount = ($_SESSION['cart']->total * $rate * $multiplier);
-}
-$ec_button_enabled = ($paypalec_enabled && $_SESSION['cart']->count_contents() > 0 && $_SESSION['cart']->total > 0 && $amount <= 10000);
-
+require_once DIR_FS_CATALOG . DIR_WS_MODULES . 'payment/paypal/paypal_currency_check.php';
+$ec_button_enabled = ($paypalec_enabled && $_SESSION['cart']->count_contents() > 0 && $_SESSION['cart']->total > 0 && paypalUSDCheck($_SESSION['cart']->total));
```

Comment:
Since that `paypalUSDCheck` returns a boolean, my preference (non-binding!) would be to check for an exactly-equal-to on the return.

Reply:
Yes agreed. It is clearer and easier to see.